server/controllers/ArticleController.js

```diff
@@ -0,0 +1,252 @@
+import slugCreate from 'slug';
+import uuid from 'uuid/v1';
+import models from '../database/models';
+
+const { Article, User } = models;
+
+/**
+ *
+ * The ArticleController contains static methods that are used as
+ * controllers to handle the different article routes.
+ *  @class ArticleController
+ *  @returns {undefined} this is a class thus does not return anything
+ */
+class ArticleController {
+  /**
+   * Method adds a new article to the database
+   * It returns a json response of the article created
+   * It handles the POST /api/articles.
+   * @param {Object} req the request body
+   * @param {Object} res the response body
+   * @param {function} next a call to the next function
+   * @returns {Object} the response body
+   */
+  static createArticle(req, res, next) {
+    const {
+      title, description, body,
+    } = req.body.article;
+
+      // generating a slug
+    const id = uuid();
+    const slug = slugCreate(`${title.toLowerCase()}-${id}`);
+
+    const authorId = req.body.decoded.id;
+
+    User.findOne({
+      where: { id: authorId },
+    })
+      .then((user) => {
+        if (user) {
+          user.createUserArticle({
+            id, slug, title, description, body,
+          })
+            .then((article) => {
+              res.status(201).json({
+                article: {
+                  id: article.id,
+                  title: article.title,
+                  description: article.description,
+                  body: article.body,
+                  slug,
+                  image: article.image,
+                  createdAt: article.createdAt,
+                  updatedAt: article.updatedAt,
+                  author: {
+                    id: user.id,
+                    username: user.username,
+                  },
+                },
+                message: 'Article Sucessfully Created',
+              });
+            })
+            .catch(next);
+        } else {
```

Comment:
Instead of nesting large blocks of code inside `if` statements, you can check if a user does not exist first, like so; `if (!user)` return the error response, then beneath it write your code for when the user exists. You can do this for the rest of your controllers.

Reply:
done

Reply:
done